### PR TITLE
[complex32, jiterator] tan, atan

### PR DESCRIPTION
**Pull request by khushi-411**
Follows #74537 and #74748

cc @kshitij12345

### CODE REVIEW

**Comment by khushi-411** [comment]
Hi, @kshitij12345! Gentle ping :)

**Review by kshitij12345 — APPROVED** [review]
LGTM! Thanks @khushi-411 

cc: @anjali411

**Comment by khushi-411** [comment]
Hi, @anjali411! Gentle ping :)

File: aten/src/ATen/native/cuda/UnaryGeometricKernels.cu [hunk 1]
```diff
@@ -131,15 +158,42 @@ void atanh_kernel_cuda(TensorIteratorBase& iter) {
       });
 }
 
+const char tan_name[] = "tan_kernel";
```
Comment (ngimel) [inline aten/src/ATen/native/cuda/UnaryGeometricKernels.cu:161]: nit for the future - currently `_kernel` is unconditionally added to the kernel name by codegen, so this kernel will show up in the profile as `tan_kernel_kernel`. If possible, don't add `kernel` to names in ops (this is a preexisting condition for a lot of existing kernels, so no need to fix now).

Comment (khushi-411) [reply aten/src/ATen/native/cuda/UnaryGeometricKernels.cu:161]: Hi, @ngimel!
Noted. I'll keep that in my mind. Thanks!

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -18631,6 +18656,14 @@ def __init__(
     ElementwiseUnaryPythonRefInfo(
         "_refs.atan",
         torch_opinfo_name="atan",
+        skips=(
+            # RuntimeError: "index_select_cuda" not implemented for 'ComplexHalf'
+            DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_python_reference_consistency',
+                         dtypes=(torch.chalf,)),
+            # same reason as above
+            DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_python_reference_meta_functions',
```
[the remark this answers is not in the document]
Comment (khushi-411) [reply torch/testing/_internal/common_methods_invocations.py:18864]: Sure, I've rebased and removed that. Thanks!

**Review by ngimel — APPROVED** [review]
Cool, let's wait for CI

**Pull request closed without merging**